### DISCUSSION TITLE
[Feature] I hope you can customize the upload location for uploading photos

### DISCUSSION THREAD

**Discussion by pyccl** [immich-app/immich#12523] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When uploading, you can choose a directory with permission to upload. You can directly upload to the folder on the 'Folder' page. Our current storage is folder by folder, and immich seems to be random like most. The specific location where this file is archived is unknown to most people.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by pyccl** [comment]
It can also be uploaded to an external gallery function. Because many people, like me, have a photo of a tourist attraction, a folder, a folder of a place, and so on

**Comment by bo0tzz** [comment]
You can customize the upload folder stucture with storage template https://immich.app/docs/administration/storage-template

**Reply by pyccl** [reply]
This seems to be a simple storage. My photos may not always exist in any folder, for example, I have some folders such as "flowers", "birds", "parks", "zoos", "amusement parks", etc. These folders are not in the same directory, such as "Beijing/Zoo", "Beijing/Botanical Garden", "Shanghai/Disney", "Hong Kong/University", etc. The photos I want to upload are from these directories. How should we do this? I feel like I can't do it yet, can I? So I hope that when uploading, I can personally specify a directory or create a new directory.

**Reply by bo0tzz** [reply]
I don't understand what you're asking for. The storage template can include the album an asset is in, if that's what you mean.

**Reply by pyccl** [reply]
I am used to managing photos in a folder format, similar to an album. One album, one folder. Using nested folder management (similar to photo albums and sub albums) to store templates seems to require setting up templates in advance when uploading. And the upload only goes to the designated storage template below. What I want is to choose a folder (or sub folder) to upload photos directly to the folder, so that I can also manage them using Windows. My English is not good, this was translated using a translation website. I don't know if you understand.
